src/smt/smt_engine.cpp

```diff
@@ -5647,6 +5689,19 @@ void SmtEngine::setReplayStream(ExprStream* replayStream) {
   AlwaysAssert(!d_fullyInited,
                "Cannot set replay stream once fully initialized");
   d_replayStream = replayStream;
+}  
+
+std::map< Expr, std::string > SmtEngine::getExpressionNames() const {
+  return d_private->getExpressionNames();
+}
+
+bool SmtEngine::getExpressionName(Expr e, std::string& name) const {
```

Comment:
Suggestion: This method could also return an `std::pair<bool, std::string>`, then the user can use [std::tie](http://en.cppreference.com/w/cpp/utility/tuple/tie) to conveniently access the two results.

Reply:
I prefer this style more, for another example, see:
https://github.com/CVC4/CVC4/blob/master/src/theory/valuation.h#L96

Reply:
Sure, it is mostly a matter of personal preference.